sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuCompressedColumnVector.java

```diff
@@ -16,151 +16,66 @@
 
 package com.nvidia.spark.rapids;
 
-import ai.rapids.cudf.DType;
 import ai.rapids.cudf.DeviceMemoryBuffer;
-import com.nvidia.spark.rapids.format.ColumnMeta;
 import com.nvidia.spark.rapids.format.TableMeta;
-import org.apache.spark.sql.types.ArrayType;
-import org.apache.spark.sql.types.BinaryType;
-import org.apache.spark.sql.types.DataType;
-import org.apache.spark.sql.types.MapType;
-import org.apache.spark.sql.types.StructField;
-import org.apache.spark.sql.types.StructType;
 import org.apache.spark.sql.vectorized.ColumnVector;
 import org.apache.spark.sql.vectorized.ColumnarBatch;
 
+import static org.apache.spark.sql.types.DataTypes.NullType;
+
 /**
  * A GPU column vector that has been compressed. The columnar data within cannot
  * be accessed directly. This class primarily serves the role of tracking the
  * compressed data and table metadata so it can be decompressed later.
  */
-public final class GpuCompressedColumnVector extends GpuColumnVectorBase {
+public final class GpuCompressedColumnVector extends GpuColumnVectorBase
+    implements WithTableBuffer {
+  private static final String BAD_ACCESS_MSG = "Column is compressed";
+
   private final DeviceMemoryBuffer buffer;
   private final TableMeta tableMeta;
 
   /**
    * Build a columnar batch from a compressed table.
    * NOTE: The data remains compressed and cannot be accessed directly from the columnar batch.
    */
-  public static ColumnarBatch from(CompressedTable compressedTable, DataType[] colTypes) {
-    return from(compressedTable.buffer(), compressedTable.meta(), colTypes);
+  public static ColumnarBatch from(CompressedTable compressedTable) {
+    return from(compressedTable.buffer(), compressedTable.meta());
   }
 
   public static boolean isBatchCompressed(ColumnarBatch batch) {
-    if (batch.numCols() == 0) {
+    if (batch.numCols() != 1) {
```

Comment:
consider making this method a one-liner 
```suggestion
    return batch.numCols() == 1 && batch.column(0) instanceof GpuCompressedColumnVector
```